static/js/components/Navigation.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import React from "react"
+import R from "ramda"
+import { Link } from "react-router-dom"
+
+import SubscriptionsList from "./SubscriptionsList"
+
+import { newPostURL } from "../lib/url"
+
+import type { Channel } from "../flow/discussionTypes"
+
+export const getChannelName = R.compose(R.defaultTo(null), R.view(R.lensIndex(1)), R.match(/^\/channel\/([a-z]+)\/?/))
```

Comment:
I dont *think* that I can use `getChannelName`, although it could be because I am not correctly understanding how react router works now. But the issue I ran into was that, because this `Navigation` component is rendered at a few different points in the routing hierarchy (at the top level for the navigation drawer, and inside of a route for the pages like the frontpage, channel page, etc) the same URL parameters will not always be there.

The thing I noticed was basically this: if the `window.location` was set to something like `/channel/foobarchannel`, the `Navigation` component inside of the `ChannelPage` container would receive a `Match` prop that had the `channelName` URL parameter parsed out of it correctly. I had thought that this would be true globally, but it is not, unfortunately. It seems to only happen because the `<ChannelPage />` is being rendered inside of a 
```
<Route path={`${match.url}channel/:channelName`} />
```
 Since the `Navigation` component for the `Drawer` has to be rendered at the top-level of the app, it isn't inside such a route, so the `Match` prop that it receives doesn't have this URL parameter parsed out.

Does that make sense? This is the simplest workaround that I found, basically relying on the fact that at least the `location.pathname` is going to be the same regardless of where in the routing hierarchy the component is being rendered.

Reply:
OK. I guess we're working around some library quirks here which makes it more difficult, but at the very least I think we should create a util method called `getSelectedChannelFromUrl` or something like that. I also think `submitPostButton` should just take a string for the channel name. I don't think the button-rendering function should worry about finding the channel name itself.

Reply:
OK, I'll move the function to another file and do a little refactoring